tests/check_warnings.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+import sys
+
+from colorama import Fore, init
+
+# init colors for all plateforms
+init()
+
+
+def check_warnings(file):
+    """
+    Check the list of warnings produced by the GitHub CI tests
+    raise errors if there are unexpected ones and/or if some are missing
+
+    Args:
+        file (pathlib.Path): the path to the generated warning.txt file from
+            the CI build
+
+    Return:
+        0 if the warnings are all there
+        1 if some warning are not registered or unexpected
+    """
+
+    # print some log
+    print("\n=== Sphinx Warnings test ===\n\n")
```

Comment:
I think it'd be helpful if we printed a message like "Checking build warnings in file: {file} and comparing to expected warnings defined in {path-to-expected-warnings}". That should make this easier to debug if people come across a failure here and don't know where to look